DataDog scaler makes KEDA operator crashing loop back

### Report

I recently tried to use DataDog Scaler to scale my deployment. But every time, when i apply the scaledobjects like below, the keda-operator pod will crash loopback with a stack trace error

*Keda version 2.6.0*

Trying to apply the simple manifest like 

```apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: keda-demo-4
  namespace: development
spec:
  minReplicaCount: 1
  maxReplicaCount: 20
  scaleTargetRef:
    name: supper-app
  triggers:
  - type: datadog
    metadata:
      # Required: datadog metric query
      query: "max:aws.sqs.approximate_number_of_messages_visible"
      # Required: according to the number of query result, to scale the TargetRef
      queryValue: "10"
      # Optional: (Global or Average). Whether the target value is global or average per pod. Default: Average
      type: "Global"
      # Optional: The time window (in seconds) to retrieve metrics from Datadog. Default: 90
      age: "60"
    authenticationRef:
      name: keda-trigger-auth-datadog-secret
```

The Keda Operator pod will throw the error and keep crashing loopback

![image](https://user-images.githubusercontent.com/57263227/153658761-4723aa9a-a3cc-4f1d-97df-b19898fb40a8.png)

Trying to hit API DataDog, the metrics is existing
![image](https://user-images.githubusercontent.com/57263227/153659517-17ac60da-2780-4b39-8b10-792f54ddc808.png)



### Expected Behavior

DataDog scaler should run properly. 

### Actual Behavior

Keda Operator will be crashed and hung forever until We delete that scaledObjects (keda-demo-4)

### Steps to Reproduce the Problem

1. Deploy Keda as usual (version 2.60)
2. Apply scaledObjects like above

### Logs from KEDA operator

```
panic: runtime error: slice bounds out of range [:-1]

goroutine 476 [running]:
github.com/kedacore/keda/v2/pkg/scalers.parseDatadogMetadata(0xc0010d71e0)
        /workspace/pkg/scalers/datadog_scaler.go:130 +0x645
github.com/kedacore/keda/v2/pkg/scalers.NewDatadogScaler({0x3781978, 0xc000669800}, 0x37ca550)
        /workspace/pkg/scalers/datadog_scaler.go:48 +0x2f
github.com/kedacore/keda/v2/pkg/scaling.buildScaler({0x3781978, 0xc000669800}, {0x37ca550, 0xc000e27770}, {0xc000922ba4, 0x7}, 0x37a7860)
        /workspace/pkg/scaling/scale_handler.go:383 +0x258
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).buildScalers.func1()
        /workspace/pkg/scaling/scale_handler.go:322 +0x365
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).buildScalers(0xc0006217a0, {0x3781978, 0xc000669800}, 0xc0000e0500, 0xc00061b200, {0x0, 0x0})
        /workspace/pkg/scaling/scale_handler.go:325 +0x5dc
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).GetScalersCache(0xc0006217a0, {0x3781978, 0xc000669800}, {0x307ede0, 0xc000dce000})
        /workspace/pkg/scaling/scale_handler.go:194 +0x2fa
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).getScaledObjectMetricSpecs(0xc001247d40, {0x3781978, 0xc000669800}, {{0x37a7860, 0xc000669830}, 0xc0010d77e0}, 0xc000dce000)
        /workspace/controllers/keda/hpa.go:163 +0x8c
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).newHPAForScaledObject(0xc001247d40, {0x3781978, 0xc000669800}, {{0x37a7860, 0xc000669830}, 0xc000f3b580}, 0xc000dce000, 0xc0010d7a48)
        /workspace/controllers/keda/hpa.go:63 +0x66
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).createAndDeployNewHPA(0xc001247d40, {0x3781978, 0xc000669800}, {{0x37a7860, 0xc000669830}, 0xc0005ef740}, 0xc000dce000, 0x3810b28)
        /workspace/controllers/keda/hpa.go:46 +0x216
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).ensureHPAForScaledObjectExists(0xc001247d40, {0x3781978, 0xc000669800}, {{0x37a7860, 0xc000669830}, 0x37a7860}, 0xc000dce000, 0x0)
        /workspace/controllers/keda/scaledobject_controller.go:361 +0x2d0
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).reconcileScaledObject(0xc001247d40, {0x3781978, 0xc000669800}, {{0x37a7860, 0xc000669830}, 0x0}, 0xc000dce000)
        /workspace/controllers/keda/scaledobject_controller.go:229 +0x1ae
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).Reconcile(0xc001247d40, {0x3781978, 0xc000669800}, {{{0xc000922b40, 0x2f4b9a0}, {0xc000922b30, 0x30}}})
        /workspace/controllers/keda/scaledobject_controller.go:180 +0x38a
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc000fa8dc0, {0x3781978, 0xc000669740}, {{{0xc000922b40, 0x2f4b9a0}, {0xc000922b30, 0x413974}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000fa8dc0, {0x37818d0, 0xc0012bd180}, {0x2d15e20, 0xc00007ee60})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000fa8dc0, {0x37818d0, 0xc0012bd180})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x357
```


### KEDA Version

2.6.0

### Kubernetes Version

1.21

### Platform

Other

### Scaler Details

DataDog.

### Anything else?

In case i changed the DataDog query with an Tag filter like 
`max:aws.sqs.approximate_number_of_messages_visible{app_name:supper-app}`

Then the keda-operator will run properly however that metric will not be able to return along with the error (from keda-metric-server) 

```
E0211 16:12:52.946301       1 provider.go:124] keda_metrics_adapter/provider "msg"="error getting metric for scaler" "error"="error getting metrics from Datadog: no Datadog metrics returned"  "scaledObject.Name"="keda-demo-4" "scaledObject.Namespace"="development" "scaler"={}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,7 @@
 ### Improvements
 
 - **Azure Queue:** Don't call Azure queue GetProperties API unnecessarily ([#2613](https://github.com/kedacore/keda/pull/2613))
+- **Datadog Scaler:** Validate query to contain `{` to prevent panic on invalid query ([#2625](https://github.com/kedacore/keda/issues/2625))
 
 ### Breaking Changes
 

--- a/pkg/scalers/datadog_scaler.go
+++ b/pkg/scalers/datadog_scaler.go
@@ -68,6 +68,9 @@ func parseDatadogMetadata(config *ScalerConfig) (*datadogMetadata, error) {
 	} else {
 		return nil, fmt.Errorf("no query given")
 	}
+	if !strings.Contains(meta.query, "{") {
+		return nil, fmt.Errorf("expecting query to contain `{`, got %s", meta.query)
+	}
 
 	if val, ok := config.TriggerMetadata["queryValue"]; ok {
 		queryValue, err := strconv.Atoi(val)

--- a/pkg/scalers/datadog_scaler_test.go
+++ b/pkg/scalers/datadog_scaler_test.go
@@ -41,6 +41,8 @@ var testDatadogMetadata = []datadogAuthMetadataTestData{
 	{map[string]string{"query": "sum:trace.redis.command.hits{env:none,service:redis}.as_count()", "queryValue": "7"}, map[string]string{"appKey": "appKey"}, true},
 	// missing appKey
 	{map[string]string{"query": "sum:trace.redis.command.hits{env:none,service:redis}.as_count()", "queryValue": "7"}, map[string]string{"apiKey": "apiKey"}, true},
+	// invalid query missing {
+	{map[string]string{"query": "sum:trace.redis.command.hits.as_count()", "queryValue": "7"}, map[string]string{}, true},
 }
 
 func TestDatadogScalerAuthParams(t *testing.T) {

